javascript.md

```diff
@@ -1121,7 +1257,64 @@
 -
   <details>
     <summary>
-      <b>A15.</b> Conditions are simplified.
+      <b>A15.1.</b> Where possible, an object is used instead of a `switch` statement.
+    </summary>
+
+    Bad:
+
+    ```typescript
+      const UserStatuses = {
+        active = 'active',
+        paused = 'paused',
+        deleted = 'deleted'
+      }
+
+      const getIndicatorColorFromUserStatus  = (status) => {
+        switch(status){
+          case UserStatuses.active: {
+            return 'green'
+          }
+          case UserStatuses.paused: {
+            return 'yellow'
+          }
+          case UserStatuses.deleted: {
+            return 'grey'
+          }
+          default: {
+            throw new Error(`Unknown status: ${status}`)
+          }
+        }
+      }
+    ```
+  
+    Good:
+  
+    ```typescript
+      const UserStatuses = {
+        active = 'active',
+        paused = 'paused',
+        deleted = 'deleted'
+      }
+
+      const UserStatusIndicatorColor = {
+        [UserStatuses.active]: 'green',
+        [UserStatuses.paused]: 'yellow',
+        [UserStatuses.deleted]: 'grey'
+      }
```

Comment:
```suggestion
      const userStatusToColor = {
        [UserStatus.ACTIVE]: 'green',
        [UserStatus.PAUSED]: 'yellow',
        [UserStatus.DELETED]: 'grey'
      }
```
Looks like map data structure. In JS world it has some conventions for naming and use